external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchStream.scala

```diff
@@ -98,7 +100,10 @@ private[kafka010] class KafkaMicroBatchStream(
     } else if (minOffsetPerTrigger.isDefined) {
       ReadLimit.minRows(minOffsetPerTrigger.get, maxTriggerDelayMs)
     } else {
-      maxOffsetsPerTrigger.map(ReadLimit.maxRows).getOrElse(super.getDefaultReadLimit)
+      // Calling ReadLimit.allAvailable() explicitly instead of super.getDefaultReadLimit because
+      // scala compiler bug https://github.com/scala/bug/issues/12523
+      // TODO revert to calling super.getDefaultReadLimit when scala complier bug is fixed
```

Comment:
It is better to create a JIRA ticket and add into the comment here for tracking it.